WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -77,6 +79,18 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     var checkoutButtonDisabled: Bool {
         return itemsInCart.isEmpty
     }
+
+    func cardPaymentTapped() {
```

Comment:
Curious to get your thoughts to make it work for the latest design (likely the final):

<img width="336" alt="Screenshot 2024-06-05 at 1 50 57 PM" src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/d2431613-b99b-46c3-969a-bd85fcbeda9c">

In this design, there's no CTA to take card payment if the reader is already connected (I'd think there is one CTA if the reader isn't connected). In this case (reader connected before landing on this checkout screen), do you think we'd just call `cardPresentPaymentService.collectPayment` right after knowing that the reader is connected on the checkout view? It's a good thing that we can decide how to show the UI now.

Reply:
Yes, exactly.

If the reader's connected, we start the payment when the screen loads and show a relevant message where "Tap or insert card to pay" is shown – I'd say we keep it simple and generally just say "preparing reader" until it's ready.

If the reader's not connected, I think we should show that flow modally to start with – but we could choose to only pop up the "connect to this reader" modal which they have to say yes or no to. Or show the buttons in-line.

Errors – tougher, as the merchant might be planning to take cash anyway, in which case it's annoying to have a card payment error pop up... but if they're taking a card payment, it needs to be obvious if that's failed for some reason, so showing a modal might be better there.